### DISCUSSION TITLE
Mid-game ship editing

### DISCUSSION THREAD

**Discussion by Quartofel** [FakeFishGames/Barotrauma#7476] [Ideas and suggestions]
Add an option to customize sub to some degree without the need to buy a completely new sub.
Sometimes, all you need to make your sub quality of life skyrocket is just one more ammo rack near the loaders, yet, such feat is impossible without purchasing entire new submarine. BaroCraftables mod is a great example on how much such feature improves the quality of life for players, albeit some limitations for placement of addional items would be nice to have for balance's sake.

Potentially hull-alterations could be added as well on dedicated shipyard outposts.

Such feature also adds to the immersion, especially during longplay campaigns with friends, since during your seafaring you can grow attached to your good ol' dugong that somehow survived two danger-close encounters with the worm boi.

**Comment by MaaiG** [comment]
> since during your seafaring you can grow attached to your good ol' dugong that somehow survived two danger-close encounters with an worm boi.

👍 
The weapons installation mechanism has been added to the game: `Periscope Base`, `Loader Base`. I think vanilla submarines just haven’t kept up with the changes.

**Reply by Quartofel** [reply]
Vanilla subs have weapon bases, I'm talking more about adding things like tables, cabinets, shelves etc. Adding weapon hardpoints or modifying hulls could be cool, but only on dedicated shipyard outposts.

**Comment by mLuby** [comment]
Possible duplicate of ["Utility Hardpoints": An extra step in sub progression and variation #6180](https://github.com/Regalis11/Barotrauma/discussions/6180).

Until this suggestion gets implemented, the online [baro-save-editor](https://ignis05.github.io/baro-save-editor/) makes editing in a new version of your sub simpler.

Also one of the devs said the following:
> If I do a campaign and then type `subeditor` in the `f3` console. Do I have to save my game or anything? All changes made in sub editor appear after I type `game`, I assume? -[a player](https://discord.com/channels/103209684680323072/157565009726144513/905866139064762418)

> Just exit the editor with `game` command and the changes should already be there. Just remember that it's not an official feature; you may break everything. -[a dev](https://discord.com/channels/103209684680323072/157565009726144513/905866256622702602)

**Comment by gerakis** [comment]
I would also like to make changes to in game subs. In the Berilia I added a door to cargo, a bunch of storage and moved the cargo spawn point to the fabricator room. It worked great! Just a few simple changes made the ship way less annoying.
